app/src/main/java/fr/free/nrw/commons/utils/ContributionUtils.java

```diff
@@ -47,10 +48,18 @@ public static Uri saveFileBeingUploadedTemporarily(Context context, Uri URIfromC
         return result;
     }
 
-    public static void removeTemporaryFile(Context context, Uri tempFileUri) {
+    /**
+     * Removes temp file created during upload
+     * @param context
+     * @param tempFileUri
+     * @param contentProviderUri
+     */
+    public static void removeTemporaryFile(Context context, Uri tempFileUri, Uri contentProviderUri) {
+        //TODO: do I have to notify file system about deletion?
```

Comment:
Am not sure about notifying to the file system but our contributionDao should be updating the table if it has a corresponding tuple.

Reply:
We only delete temp file here, after contribution is uploaded. If we delete contribution from contributionDao, then our successful contribution will be removed from contributions table. Then it won't be visible on contribution list.